### PR TITLE
Envelope trigger

### PR DESCRIPTION
**Pull request by lpyras**
Issue being addressed:

Did you remember to: 
* include doc strings in all commits ?
* edit the changelog.txt file describing your changes?

### CODE REVIEW

File: NuRadioReco/modules/trigger/envelopeTrigger.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+    def begin(self, debug=False):
+        self.__debug = debug
+
+
+    @register_run()
+    def run(self, evt, station, det, passband, order, threshold, coinc_window, number_coincidences=2, triggered_channels=None,
+             trigger_name='envelope_trigger'):
+        """
+        simulate simple trigger logic, no time window, just threshold in all channels
```
Comment (anelles) [inline NuRadioReco/modules/trigger/envelopeTrigger.py]: Doumentation looks outdated. There should be time windows, correct? Please update.

File: NuRadioReco/examples/FullReconstructionMultiFiles.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+detector_file: str
+            path to json detector database, default given
+"""
+
+try:
+    station_id = int(sys.argv[1])  # specify station id
+    input_files = sys.argv[2].split(',')  # file with coreas simulations
+
+except:
+    print("Usage: python FullReconstruction.py station_id input_file detector")
```
[the remark this answers is not in the document]
Comment (lpyras) [reply NuRadioReco/examples/FullReconstructionMultiFiles.py]: I removed the FullReconstructionMultiFiles.py. It is very similar to FullReconstruction and may confuse people.

File: NuRadioReco/framework/trigger.py [hunk 3]
```diff
@@ -251,3 +254,34 @@ def __init__(self, name, threshold, channel_coincidence_window, channels=None, n
...
+    def __init__(self, name, passband, order, threshold, number_of_coincidences=2,
+                 channel_coincidence_window=None, channels=None):
+        """
+        initialize trigger class
+
+        Parameters
+        -----------
+        name: string
+            unique name of the trigger
+        passband:
```
[the remark this answers is not in the document]
Comment (lpyras) [reply NuRadioReco/framework/trigger.py]: Done!

File: NuRadioReco/modules/trigger/envelopeTrigger.py (hunk 1 above)
Comment (lpyras) [reply NuRadioReco/modules/trigger/envelopeTrigger.py]: Done.

File: NuRadioReco/modules/trigger/envelopeTrigger.py [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+                channel_trace_start_time = channel.get_trace_start_time()
+                break
+        else:
+            channel_trace_start_time = station.get_channel(triggered_channels[0]).get_trace_start_time()
+
+        event_id = evt.get_id()
+        for channel in station.iter_channels():
+            # get filter
+            frequencies = channel.get_frequencies()
+            frequencies_tri = copy.copy(frequencies)
```
[the remark this answers is not in the document]
Comment (lpyras) [reply NuRadioReco/modules/trigger/envelopeTrigger.py]: True.

File: NuRadioReco/modules/trigger/envelopeTrigger.py [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+            f = np.zeros_like(frequencies_tri, dtype=np.complex)
+            mask = frequencies_tri > 0
+            b, a = scipy.signal.butter(order, passband, 'bandpass', analog=True)
+            w, h = scipy.signal.freqs(b, a, frequencies_tri[mask])
+            f[mask] = h
+            print(f)
+
+            # apply filter
+            freq_spectrum_fft = channel.get_frequency_spectrum()
+            freq_spectrum_fft_tri = copy.copy(freq_spectrum_fft)
```
[the remark this answers is not in the document]
Comment (lpyras) [reply NuRadioReco/modules/trigger/envelopeTrigger.py]: Here, I need it, so the spectrum is only changed within the trigger module. I made a commit in the code.

File: NuRadioReco/modules/trigger/envelopeTrigger.py [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+            # get filter
+            frequencies = channel.get_frequencies()
+            frequencies_tri = copy.copy(frequencies)
+
+            f = np.zeros_like(frequencies_tri, dtype=np.complex)
+            mask = frequencies_tri > 0
+            b, a = scipy.signal.butter(order, passband, 'bandpass', analog=True)
+            w, h = scipy.signal.freqs(b, a, frequencies_tri[mask])
+            f[mask] = h
+            print(f)
```
[the remark this answers is not in the document]
Comment (lpyras) [reply NuRadioReco/modules/trigger/envelopeTrigger.py]: I changed it to logger.info.

**Pull request merged**